### PR TITLE
proxy kernels updates kernel info on first command

### PR DESCRIPTION
**Pull request by colombod**
image

Now javascript kernel language is correctly used in the proxy KernelInfo

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs [hunk 1]
```diff
@@ -34,24 +34,22 @@ public sealed class ProxyKernel : Kernel
 
         if (remoteUri is not null)
         {
-            _remoteUri = remoteUri;
+            KernelInfo.RemoteUri = remoteUri;
         }
         else if (sender.RemoteHostUri is { } remoteHostUri)
         {
-            _remoteUri = new(remoteHostUri, name);
+            KernelInfo.RemoteUri = new(remoteHostUri, name);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs:41]: I wonder if the bug we were looking at in the browser kernel yesterday means this the wrong behavior. It sort of works by accident, sometimes, but it's not really a safe assumption.

File: src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs [hunk 2]
```diff
@@ -121,6 +123,25 @@ private Task HandleByForwardingToRemoteAsync(KernelCommand command, KernelInvoca
         });
     }
 
+    private void EnsureKernelInfoIsLoaded(KernelCommand command, KernelInvocationContext context)
+    {
+        if (_requiresRequestKernelInfoOnFirstCommand)
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs:128]: Can this be done using a deferred command?

Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs:128]: Though I guess that would produce a doubled `RequestKernelInfo` in some cases...

File: src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs [hunk 3]
```diff
@@ -184,6 +205,10 @@ private void DelegatePublication(KernelEvent kernelEvent)
                     _inflight.Remove(token);
                     pending.completionSource.TrySetResult(cs);
                     break;
+                case CommandFailed _ when kernelEvent.Command.GetOrCreateId() == _suppressCompletionsForCommandId:
+                case CommandSucceeded _ when kernelEvent.Command.GetOrCreateId() == _suppressCompletionsForCommandId:
+                    _suppressCompletionsForCommandId = null;
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs:210]: Why is this needed?

File: src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernel.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ internal class MsSqlKernel : ToolsServiceKernel
         internal MsSqlKernel(
             string name,
             string connectionString,
-            ToolsServiceClient client) : base(name, client, "SQLite")
+            ToolsServiceClient client) : base(name, client, "SQL")
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernel.cs:23]: `T-SQL`?

Comment (colombod) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernel.cs:23]: @brettfo  what are you thoughts on this language

Comment (brettfo) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernel.cs]: "SQL" is good; it'll play well with my kernel picker work

**Review by brettfo — APPROVED** [review]

**Comment by colombod** [comment]
Mssqlserver language is t-sql, SQLite is SQLite not sure how we want to handle that part. I have set the discovery kernel to be sql Dr Diego Colombo PhDOn 27 Oct 2022, at 21:50, Brett V. Forsgren ***@***.***> wrote:﻿
@brettfo commented on this pull request.



In src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernel.cs:
> @@ -20,7 +20,7 @@ internal class MsSqlKernel : ToolsServiceKernel
         internal MsSqlKernel(
             string name,
             string connectionString,
-            ToolsServiceClient client) : base(name, client, "SQLite")
+            ToolsServiceClient client) : base(name, client, "SQL")

"SQL" is good; it'll play well with my kernel picker work
